From f57aaa85465ad001119acfdfffc656bfa8b9acb7 Mon Sep 17 00:00:00 2001
From: J. Bruce Fields <bfields@redhat.com>
Date: Tue, 16 Sep 2014 20:18:09 -0400
Subject: [fs] nfsd: add __force to opaque verifier field casts

Message-id: <1410898849-22126-6-git-send-email-bfields@redhat.com>
Patchwork-id: 95690
O-Subject: [Patch RHEL7 005/165] nfsd: add __force to opaque verifier field casts
Bugzilla: 1044798 1123721 1141932
RH-Acked-by: Benjamin Coddington <bcodding@redhat.com>
RH-Acked-by: Steve Dickson <SteveD@redhat.com>

From: Jeff Layton <jlayton@primarydata.com>

sparse complains that we're stuffing non-byte-swapped values into
__be32's here. Since they're supposed to be opaque, it doesn't matter
much. Just add __force to make sparse happy.

Signed-off-by: Jeff Layton <jlayton@primarydata.com>
Reviewed-by: Christoph Hellwig <hch@lst.de>
Signed-off-by: J. Bruce Fields <bfields@redhat.com>
Upstream: f419992c1f792f2ce501585853ffc71b8f78caa1
Bugzilla: 1044798
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/nfsd/nfs4proc.c b/fs/nfsd/nfs4proc.c
index 3ece7cd..53ac6bd 100644
--- a/fs/nfsd/nfs4proc.c
+++ b/fs/nfsd/nfs4proc.c
@@ -580,8 +580,12 @@ static void gen_boot_verifier(nfs4_verifier *verifier, struct net *net)
  __be32 verf[2];
  struct nfsd_net *nn = net_generic(net, nfsd_net_id);
 
- verf[0] = (__be32)nn->nfssvc_boot.tv_sec;
- verf[1] = (__be32)nn->nfssvc_boot.tv_usec;
+ /*
+  * This is opaque to client, so no need to byte-swap. Use
+  * __force to keep sparse happy
+  */
+ verf[0] = (__force __be32)nn->nfssvc_boot.tv_sec;
+ verf[1] = (__force __be32)nn->nfssvc_boot.tv_usec;
  memcpy(verifier->data, verf, sizeof(verifier->data));
 }
 
diff --git a/fs/nfsd/nfs4state.c b/fs/nfsd/nfs4state.c
index 2204e1f..8242385 100644
--- a/fs/nfsd/nfs4state.c
+++ b/fs/nfsd/nfs4state.c
@@ -1421,8 +1421,12 @@ static void gen_confirm(struct nfs4_client *clp)
  __be32 verf[2];
  static u32 i;
 
- verf[0] = (__be32)get_seconds();
- verf[1] = (__be32)i++;
+ /*
+  * This is opaque to client, so no need to byte-swap. Use
+  * __force to keep sparse happy
+  */
+ verf[0] = (__force __be32)get_seconds();
+ verf[1] = (__force __be32)i++;
  memcpy(clp->cl_confirm.data, verf, sizeof(clp->cl_confirm.data));
 }
 
-- 
1.7.1